src/coreclr/tools/superpmi/superpmi/superpmi.cpp

```diff
@@ -552,16 +572,42 @@ int __cdecl main(int argc, char* argv[])
                 }
                 else
                 {
-                    bool hasDiff;
-                    InvokeNearDiffer(&nearDiffer, &o, &mc, &crl, &hasDiff, &reader, &failingToReplayMCL, &diffMCL);
+                    NearDifferResult result = InvokeNearDiffer(&nearDiffer, &mc, &crl, &reader);
 
-                    if (hasDiff)
+                    switch (result)
                     {
-                        totalBaseMetrics.Overall.NumContextsWithDiffs++;
-                        totalDiffMetrics.Overall.NumContextsWithDiffs++;
+                        case NearDifferResult::SuccessWithDiff:
+                            totalBaseMetrics.Overall.NumContextsWithDiffs++;
+                            totalDiffMetrics.Overall.NumContextsWithDiffs++;
+
+                            totalBaseMetricsOpts.NumContextsWithDiffs++;
+                            totalDiffMetricsOpts.NumContextsWithDiffs++;
+
+                            // This is a difference in ASM outputs from Jit1 & Jit2 and not a playback failure
+                            // We will add this MC to the diffs info if there is one.
+                            // Otherwise this will end up in failingMCList
```

Comment:
I find this a bit puzzling. Can you say more about why we would add this to the failing MCL?

Reply:
This is just preserving the existing behavior (it was inside `InvokeNearDiffer` before).
I'm not sure why we are regarding diffs as failures when `-diffMCList` (`-diffsInfo` after this change) is unspecified.